src/zenml/models/v2/core/artifact.py

```diff
@@ -42,6 +42,11 @@ class ArtifactRequest(BaseRequest):
         title="Name of the artifact.",
         max_length=STR_FIELD_MAX_LENGTH,
     )
+    original_name: Optional[str] = Field(
```

Comment:
What happens if I create two artifacts which have the same name, but different original names? I don't think this would work as expected, no?

Shouldn't the mapping of step runs to its output artifact happen in the `StepRunOutputArtifactSchema`, which is already storing a name?

In what other occasions do we need to map information stored in the deployment to these artifacts?

Reply:
You are right that it should not belong to the `Artifact` because of the duality you mentioned, but I also don't feel it belongs to the `StepRunOutputArtifactSchema`, rather to `ArtifactVersionSchema`, thus I made changes to add it there and added extra test to cover that edge case.
I don't know about any other use cases, but this is surely the case for the new DAG with all steps/outputs shown at once.

Reply:
What is the reasoning the makes sense on the artifact version though? Isn't this something that explicitly happens on step outputs? When someone manually uploads an artifact version, there would never be an `original_name`, no? Somehow it just feels unintuitive for me right now, ideally we wouldn't even need to store this IMO. Is there any way we can get around storing this at all?

Reply:
IMO, the reasoning is that this information was the driver for the Artifact Version to receive the name if we can find it in the Artifact entity. However, it can come from various sources (static or dynamic). I don't see why this should relate to the Step Output directly and overcomplicate the schema relations while filling that info into other entities. If the artifact version has no original name, e.g. for manual save or for static names - this field is just not populated in DB and only backfilled in runtime with the artifact name itself. So, with the current implementation, it would be close to no impact if the users are still relying on static naming, or with a small overhead for dynamic names to store their original source of naming.

Reply:
Regarding the second question: I doubt that there is a good way to avoid storing something if we keep the current DAG filling logic.
What I can think of as a solution could be:
- On run start we show the outputs from the Deployment only (since no info in the Run)
- Once the step was finished ly replace that part with the data from there

I don't feel that there is any harm in doing so, but I would love to hear from @Cahllagerfeld and @znegrin on that matter.